data/countries/JP.yaml

```diff
@@ -25,90 +30,283 @@ holidays:
           en: January 3rd
           jp: 銀行休業日
         type: bank
+      01-15:
+        name:
+          en: Coming of Age Day
+          jp: 成人の日
+        active:
+          - from: 1948-07-20
+            to: 1999-12-31
+      substitutes 01-15 and if sunday then next monday:
+        substitute: true
+        name:
+          en: Coming of Age Day
+          jp: 成人の日
+        active:
+          - from: 1973-04-12
+            to: 1999-12-31
       2nd monday in January:
         name:
           en: Coming of Age Day
           jp: 成人の日
-      02-11 and if sunday then next monday:
+        active:
+          - from: 2000-01-01
+      02-11:
+        name:
+          en: Foundation Day
+          jp: 建国記念の日
+        active:
+          - from: 1967-01-01
+      substitutes 02-11 and if sunday then next monday:
         substitute: true
         name:
           en: Foundation Day
           jp: 建国記念の日
-      march equinox in +09:00 and if sunday then next monday:
+        active:
+          - from: 1973-04-12
+      02-23 and if sunday then next monday:
         substitute: true
+        name:
+          en: Emperor's Birthday
+          jp: 天皇誕生日
+        active:
+          - from: 2020-01-01
+      march equinox in +09:00:
         name:
           en: Spring Equinox Day
           jp: 春分の日
-      04-29 and if sunday then next monday:
+        active:
+          - from: 1948-07-20
+      substitutes march equinox in +09:00 and if sunday then next monday:
+        substitute: true
+        name:
+          en: Spring Equinox Day
+          jp: 春分の日
+        active:
+          - from: 1973-04-12
+      04-29:
+        name:
+          en: Emperor's Birthday
+          jp: 天皇誕生日
+        active:
+          - from: 1948-07-20
+            to: 1988-12-31
+      substitutes 04-29 and if sunday then next monday:
+        substitute: true
+        name:
+          en: Emperor's Birthday
+          jp: 天皇誕生日
+        active:
+          - from: 1973-04-12
+            to: 1988-12-31
+      '04-29 and if sunday then next monday #1':
+        substitute: true
+        name:
+          en: Greenery Day
+          jp: みどりの日
+        active:
+          - from: 1989-01-01
+            to: 2006-12-31
+      '04-29 and if sunday then next monday #2':
         substitute: true
         name:
           en: Showa Day
           jp: 昭和の日
+        active:
+          - from: 2007-01-01
+      '2019-04-30':
```

Comment:
I have a question based on curiosity about this holiday. Would you expect banks to be closed on these three days?

Reply:
Yes. In Japan, we will have 10 consecutive holidays between 2019-04-27 to 2019-05-06, and 
all banks will be closed.

ATMs work 365days/24h (including holidays) for some banks, but all bank counter is closed on the national holiday.

Reply:
Cool, thanks for the info!